### PR TITLE
Handle threshold_match_probablity 0 in predict() #2420

### PR DESCRIPTION
**Pull request by browo097302**
Updated predict_from_comparison_vectors_sqls_using_settings function to handle threshold_match_probability 0 as None.  Modified the logic in lines 100-117, this is to prevent maths errors when doing log2. 

### Type of PR

- [X] BUG
- [ ] FEAT
- [ ] MAINT
- [ ] DOC

### Is your Pull Request linked to an existing Issue or Pull Request?

Existing Issue #2420

### Give a brief description for the solution you have provided

Whenever this predict function is called it will first check for 0 and assign threshold_match_probability to None. 
Then when it is later passed into log2 it won't return a math error. 
As its a minor change I haven't changed documentation or changelog.

### PR Checklist

- [ ] Added documentation for changes  
- [ ] Added feature to example notebooks or tutorial (if appropriate) 
- [ ] Added tests (if appropriate)
- [ ] Updated CHANGELOG.md (if appropriate) 
- [X] Made changes based off the latest version of Splink
- [ ] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint_and_format.html)
- [ ] Run the [spellchecker](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/contributing_to_docs.html?h=spellch#spellchecking-docs) (if appropriate)

### CODE REVIEW

File: splink/internals/predict.py [hunk 1]
```diff
@@ -97,7 +97,9 @@ def predict_from_comparison_vectors_sqls(
         bf_terms,
         sql_infinity_expression,
     )
-
+    # Add condition to treat case of 0 as None
+    if threshold_match_probability == 0:
```
Comment (RobinL) [inline splink/internals/predict.py:101]: There's whitespace at the end of these lines at the moment hence ruff is failing.

Really the best solution  is to process using ruff, but removing the whitespace manually should do it

Comment (browo097302) [reply splink/internals/predict.py:101]: Hey, yeah sorry i thought I'd fixed that let me have another look

Comment (browo097302) [reply splink/internals/predict.py:101]: That should have done it now, it was on line 102, sorry im very new to this stuff.

**Comment by RobinL** [comment]
Thanks.  Closes https://github.com/moj-analytical-services/splink/issues/2420

**Pull request merged**